### PR TITLE
OCM-5399 | fix: Remove duplicated wait function

### PR DESCRIPTION
**Pull request by nirarg**
**What this PR does / why we need it**:
This commit also fixes [OCM-5753](https://issues.redhat.com//browse/OCM-5753):
* In case of timeout and state not "READY" return error

**Which issue(s) this PR fixes**:
Fixes [OCM-5753](https://issues.redhat.com//browse/OCM-5753), [OCM-5399](https://issues.redhat.com//browse/OCM-5399)

**Change type**
- [ ] New feature
- [x] Bug fix
- [ ] Build
- [ ] CI
- [ ] Documentation
- [ ] Performance
- [x] Refactor
- [ ] Style
- [ ] Unit tests
- [x] Subsystem tests

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.

### CODE REVIEW

File: subsystem/cluster_waiter_test.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ var _ = Describe("Cluster creation", func() {
 				}
 			`)
 
-		Expect(terraform.Apply()).To(BeZero())
+		Expect(terraform.Apply()).NotTo(BeZero())
 		resource := terraform.Resource("rhcs_cluster_wait", "rosa_cluster")
```
Comment (bardielle) [inline subsystem/cluster_waiter_test.go:179]: Does it possible to add a test that cluster is still installing and the waiter will be failed?

Comment (nirarg) [reply subsystem/cluster_waiter_test.go:179]: This is the test which reproduce the issue
The cluster wait is finished with timeout and state="waiting"
This is why added the change to expect result not to be zero

**Pull request merged**